### PR TITLE
Take-off & Land events

### PR DESCRIPTION
**Pull request by LotuxPunk**
First part, I need some sleep 🗡

### CODE REVIEW

File: src/main/java/net/tardis/mod/common/tileentity/TileEntityTardis.java [hunk 1]
```diff
@@ -567,11 +570,12 @@ public void setSpaceTimeCoordnate(SpaceTimeCoord co) {
 	}
 
 	public boolean startFlight() {
-		if (this.getDestination().equals(BlockPos.ORIGIN)) return false;
-		if (fuel <= 0.0F || !getCanFly()) {
+		TardisTakeOffEvent event = new TardisTakeOffEvent(this);
+		if (MinecraftForge.EVENT_BUS.post(event) || event.getFuel() <= 0.0F || event.getDestination() == null || !getCanFly()) {
```
Comment (Suffril) [inline src/main/java/net/tardis/mod/common/tileentity/TileEntityTardis.java]: Check both if it is Origin or Null

**Comment by LotuxPunk** [comment]
Why disallow people to travel to 0.0.0?

File: src/main/java/net/tardis/mod/common/tileentity/TileEntityTardis.java (hunk 1 above)
Comment (LotuxPunk) [reply src/main/java/net/tardis/mod/common/tileentity/TileEntityTardis.java]: Why disallow people to travel to 0.0.0?

**Comment by Suffril** [comment]
You cannot land there, it is bedrock, any attempt to land there will just result in the Tardis re-adjusting it's landing position 
And even if players got there, there wouldn't be a point, it would just result in them walking out of the Tardis into a wall

**Comment by LotuxPunk** [comment]
In my opinion we have to check if the Y is 0 not specialy for the 0.0.0 pos, but i'll do your way 😄

File: src/main/java/net/tardis/mod/common/tileentity/TileEntityTardis.java (hunk 1 above)
Comment (LotuxPunk) [reply src/main/java/net/tardis/mod/common/tileentity/TileEntityTardis.java]: Fixed

**Comment by Suffril** [comment]
@Spectre0987

**Pull request merged**